### PR TITLE
TST: Fix new warnings -> error bugs found on OSX

### PR DESCRIPTION
**Pull request by seberg**
@matthew-brett can you check if these fixups work for you?

### CODE REVIEW

**Comment by seberg** [comment]
I assume that the warning on OSX (not the nose stuff), is a correct warning that (while we probably don't test it), should be there and be ignored for testing purposes.

File: scipy/io/matlab/tests/test_streams.py [hunk 1]
```diff
@@ -27,82 +28,70 @@
...
 
 
 def test_make_stream():
-    global fs, gs, cs
-    # test stream initialization
-    assert_(isinstance(make_stream(gs), GenericStream))
-    if sys.version_info[0] < 3:
-        assert_(isinstance(make_stream(cs), cStringStream))
-        assert_(isinstance(make_stream(fs), FileStream))
+    with setup_test_file() as (fs, gs, cs):
```
Comment (seberg) [inline scipy/io/matlab/tests/test_streams.py]: These means the tempfile is created three times during testing, does not seem so bad to me.

File: scipy/sparse/linalg/eigen/arpack/tests/test_arpack.py [hunk 2]
```diff
@@ -26,21 +24,6 @@
 from scipy.linalg import svd, hilbert
 
 from scipy._lib._gcutils import assert_deallocated
-from scipy._lib._numpy_compat import suppress_warnings
-
-
-# eigs() and eigsh() are called many times, so apply a filter for the warnings
-# they generate here.
-sup = suppress_warnings()
-sup.filter(message="Single-precision types in `eigs` and `eighs`")
```
Comment (seberg) [inline scipy/sparse/linalg/eigen/arpack/tests/test_arpack.py]: I found something about the single precision version of these functions being removed in 1.10 in the logs or so, I assume these are simply not needed anymore.

**Comment by matthew-brett** [comment]
Thanks for this - yes, it does fix the errors : https://travis-ci.org/MacPython/scipy-wheels/builds/251468813

**Comment by matthew-brett** [comment]
Changes look reasonable to me.

**Comment by rgommers** [comment]
Thanks for checking @matthew-brett. @seberg could you rebase this?

**Comment by rgommers** [comment]
Merged, thanks Sebastian.

**Pull request merged**